modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]DefineAPISchemasWithinAnAPIGroup/EditAPIApplicationSchema.testcase

```diff
@@ -0,0 +1,216 @@
+	property testray.component.names = "Object";
+	property testray.main.component.name = "Headless Builder";
+
+	setUp {
+		TestCase.setUpPortalInstance();
+
+		User.firstLoginPG();
+
+		task ("Given New API Application created") {
+			ApplicationAPI.createAPIApplication(
+				baseURL = "my-app",
+				title = "My-app");
+		}
+
+		task ("And Given I go to the Schemas tab of API Application edition") {
+			APIBuilderUI.switchToRelatedEntryInEditApplication(
+				entity = "Schemas",
+				title = "My-app");
+		}
+	}
+
+	tearDown {
+		ApplicationAPI.deleteAllAPIApplication();
+
+		JSONObject.deleteAllCustomObjects(pageSize = 21);
+
+		var testLiferayVirtualInstance = PropsUtil.get("test.liferay.virtual.instance");
+
+		if (${testLiferayVirtualInstance} == "true") {
+			PortalInstances.tearDownCP();
+		}
+	}
+
+	@priority = 4
+	test CanEditSchemaNameAndDescriptionInPublishedAPIApplication {
+		property portal.acceptance = "true";
+
+		task ("And Given create an published API application with related schema") {
+			APIBuilderUI.createAPISchema(
+				confirmCreation = "true",
+				fieldDescription = "Description default",
+				name = "testSchema",
+				objectDefinitionName = "APIApplication");
+
+			Alert.viewSuccessMessage();
+
+			Click(locator1 = "APIBuilder#PUBLISH_BUTTON");
+		}
+
+		task ("And Given edit the schema in Edit API Application > Schemas") {
+			Type(
+				locator1 = "TextInput#NAME",
+				value1 = "testSchema edited");
+
+			Type(
+				locator1 = "TextInput#DESCRIPTION_FIELD",
+				value1 = "Description edited");
+		}
+
+		task ("When I publish changed name and description") {
+			Click.pauseClickAt(locator1 = "APIBuilder#PUBLISH_BUTTON");
+		}
+
+		task ("Then I can see success alert message about API schema changes were published") {
+			Alert.viewSuccessMessage();
+		}
+
+		task ("And Then I can see updated name and description in Info tab") {
+			AssertElementPresent(
+				locator1 = "TextInput#NAME",
+				value1 = "testSchema edited");
+
+			AssertElementPresent(
+				locator1 = "TextInput#DESCRIPTION_FIELD",
+				value1 = "Description edited");
+		}
+	}
+
+	@priority = 4
+	test CanEditSchemaNameAndDescriptionInUnpublishedAPIApplication {
+		property portal.acceptance = "true";
+
+		task ("And Given create an published API application with related schema") {
+			APIBuilderUI.createAPISchema(
+				confirmCreation = "true",
+				fieldDescription = "Description default",
+				name = "testSchema",
+				objectDefinitionName = "APIApplication");
+
+			Alert.viewSuccessMessage();
+
+			Click(locator1 = "APIBuilder#PUBLISH_BUTTON");
+		}
+
+		task ("And Given edit the schema in Edit API Application > Schemas") {
+			Type(
+				locator1 = "TextInput#NAME",
+				value1 = "testSchema edited");
+
+			Type(
+				locator1 = "TextInput#DESCRIPTION_FIELD",
+				value1 = "Description edited");
+		}
+
+		task ("When I save changed name and description") {
+			Click.pauseClickAt(locator1 = "APIBuilder#PUBLISH_BUTTON");
```

Comment:
We would like to test Save button here, and the one we are using is the Publish button, please help update